clusterman/aws/auto_scaling_resource_group.py

```diff
@@ -296,9 +296,27 @@ def _target_capacity(self) -> float:
 
     @classmethod
     @ttl_cache(ttl=RESOURCE_GROUP_CACHE_SECONDS)
-    def _get_resource_group_tags(cls) -> Mapping[str, Mapping[str, str]]:
+    def _get_resource_group_tags(cls, filter_tag: str = '') -> Mapping[str, Mapping[str, str]]:
         """ Retrieves the tags for each ASG """
         asg_id_to_tags = {}
+
+        if filter_tag:
+            filter = [
+                {
+                    'Name': 'key',
+                    'Values': [
+                        filter_tag,
+                    ]
+                },
+            ]
+            """If the filter tag is present switch to desribe-tag method for server side
+                filtering for performance reasons
+            """
+            for page in autoscaling.get_paginator('describe_tags').paginate(Filters=filter):
+                for instance in page['Tags']:
+                    asg_id_to_tags[instance['ResourceId']] = {filter_tag: instance['Value']}
```

Comment:
Is it ok that here we just put `instance['Value']` into the dict, but below we're building a new dict?
Is `instance['Value']` is already in the desired format?

Reply:
The output expected from_get_resource_group_tags function is in below format
```
{
  asg-id: {tag-name: tag-value}
}
```
But the output from describe-tag is in below format 

```
        {
            "ResourceId": "asg-id",
            "ResourceType": "auto-scaling-group",
            "Key": "filterTagName",
            "Value": "Filter-tag value",
            "PropagateAtLaunch": true
        }
```

Due to this, On line 317, I am getting `Instance['Value']` and then creating the required dict in 
`{filter-tag: Instance["Value"]} `  format